src/parserindexer/parse_all.py

```diff
@@ -0,0 +1,144 @@
+from jsreparser import JsreParser
+import io, ntpath
+
+
+class ParseAll(CoreNLPParser):
+
+    def __init__(self, **kwargs):
+        super(ParseAll, self).__init__(**kwargs)
+        self.jsre_parser = JsreParser(**kwargs)
+
+    def generate_example_id(self, fnbase, index, ex_id):
+        # Create a unique identifier
+        return '%s_%s_%s' % (fnbase, str(index), str(ex_id))
+
+    def generate_example(self, id, label, sentence, target_index, active_index):
+        body = ''
+        for t in sentence['tokens']:
+            # Target entity is the agent;
+            # Element entity is the target (of the relation)
+            if t['index'] == target_index:
+                entity_label = 'A'
+            elif t['index'] == active_index:
+                entity_label = 'T'
+            else:
+                entity_label = 'O'
+
+            # CoreNLP indexes starting at 1
+            body += '%d&&%s&&%s&&%s&&%s&&%s ' % (t['index'] - 1,
+                                                 t['word'],
+                                                 t['lemma'],
+                                                 t['pos'],
+                                                 t['ner'],
+                                                 entity_label)
+        # Output the example
+        example = '%s\t%s\t%s\n' % (label, id, body)
+        return example
+
+    def generate_examples(self, targets, active, relations, examples, ex_id, sentence, fnbase):
+        for i in range(0, len(targets)):
+            for j in range(0, len(active)):
+                label = 0
+                # Create a unique identifier
+                id = self.generate_example_id(fnbase, sentence['index'], ex_id)
+                ex_id += 1
+                example = self.generate_example(id, label, sentence, targets[i]['index'], active[j]['index'])
+                relations.append([targets[i], active[j], sentence])
+                examples.append(example)
+        return relations, examples, ex_id
+
+    def parse_file(self, path):
+        parsed = super(ParseAll, self).parse_file(path)
+        # Build jSRE example - creates tmp_mineral and tmp_element file
+        target_mineral = []
+        example_mineral = []
+        target_element = []
+        example_element = []
+        fn = ntpath.basename(path)
+        fnbase = fn[:fn.find('.pdf')]
+        ex_mineral_id = 0
+        ex_element_id = 0
+        print('%s: Parsing %d sentences.' % \
+              (fn, len(parsed['metadata']['sentences'])))
+        for s in parsed['metadata']['sentences']:
+            # For each pair of target+(element|mineral) entities,
+            # are they in a contains relationship?
+            # Get the relevant entities (Target, Element, Mineral)
+            targets = [t for t in s['tokens'] if t['ner'] == 'Target']
+            minerals = [t for t in s['tokens'] if t['ner'] == 'Mineral']
+            elements = [t for t in s['tokens'] if t['ner'] == 'Element']
```

Comment:
I think I'll keep these ones - just personal preference as I find list comprehensions more readable than lambda expressions.  I agree that it could go either way just fine :)